DevelopGitHub/GitHubExtension/GitHubExtension.Security.WebApi.Library/Controllers/UserController.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Web.Http;
+using System.Data.Entity;
+using GitHubExtension.Security.DAL.Infrastructure;
+using Newtonsoft.Json;
+using System.Collections.Generic;
+using Microsoft.AspNet.Identity;
+using GitHubExtension.Security.DAL.Interfaces;
+using GitHubExtension.Security.StorageModels.Identity;
+using System.Security.Claims;
+
+
+namespace GitHubExtension.Security.WebApi.Library.Controllers
+{
+
+    public class UserController:BaseApiController
+    {
+        private readonly ISecurityContext _securityContext;
+        private readonly ApplicationUserManager _userManager;
+
+
+        public UserController(
+            ISecurityContext securityContext, 
+            ApplicationUserManager userManager)
+        {
+            _securityContext = securityContext; 
+            _userManager = userManager;
+        }
+
+
+        [Route("api/user")]
+        [HttpPatch]
+        public async Task<IHttpActionResult> UpdateProject()
+        {
+            string request = await this.Request.Content.ReadAsStringAsync();
+            var repo = JsonConvert.DeserializeAnonymousType(request, new { repository = "" });
+            int repositoryId;
+
+            if(!Int32.TryParse(repo.repository, out repositoryId))
```

Comment:
So you did't interact with database and when user refresh browser all you changes is gone?